agent/taskresource/volume/dockervolume_ebs.go

```diff
@@ -0,0 +1,96 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package volume
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	apiresource "github.com/aws/amazon-ecs-agent/ecs-agent/api/resource"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+const (
+	missingEbsEntryFormart = `missing %s in the EBS config`
+)
+
+type EBSTaskVolumeConfig struct {
+	VolumeId             string `json:"volumeId"`
+	VolumeName           string `json:"volumeName"`
+	VolumeSizeGib        string `json:"volumeSizeGib"`
+	SourceVolumeHostPath string `json:"sourceVolumeHostPath"`
+	DeviceName           string `json:"deviceName"`
+	// FileSystem is optional and will be present if customers explicitly set this in their task definition
+	FileSystem string `json:"fileSystem"`
+	// DockerVolumeName is internal docker name for this volume.
+	DockerVolumeName string `json:"dockerVolumeName"`
+}
+
+// ParseEBSTaskVolumeAttachment parses the ebs task volume config value
+// from the given attachment.
+func ParseEBSTaskVolumeAttachment(ebsAttachment *ecsacs.Attachment) (*EBSTaskVolumeConfig, error) {
+	ebsTaskVolumeConfig := &EBSTaskVolumeConfig{}
+	for _, property := range ebsAttachment.AttachmentProperties {
+		if property == nil {
+			return nil, fmt.Errorf("failed to parse task ebs attachment, encountered nil property")
+		}
+		if aws.StringValue(property.Value) == "" {
```

Comment:
Discussed offline. By design, we have an initial validation for non-empty property value her, and the required properties will be validate again in func `validateEBSTaskVolumeAttachment`.

Reply:
thanks!